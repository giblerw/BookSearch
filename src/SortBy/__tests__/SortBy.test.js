import { mount } from '@vue/test-utils';
import SortByComponent from '../index.vue';

describe('SortBy.vue', () => {

    it('renders the expected HTML structure, before search', () => {
        const wrapper = mount(SortByComponent, {
            propsData: {
                query: '',
                sortBy: null
            } 
        });

        expect(wrapper.find('[id="sort-button-group"]').exists()).toBe(false);
        expect(wrapper.element).toMatchSnapshot();
    })

    it('renders the expected HTML structure, after search', () => {
        const wrapper = mount(SortByComponent, {
            propsData: { query: 'harry' } 
        });

        expect(wrapper.find('[id="sort-button-group"]').exists()).toBe(true);
        expect(wrapper.element).toMatchSnapshot();
    });
});