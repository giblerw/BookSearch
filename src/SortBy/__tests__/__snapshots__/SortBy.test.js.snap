// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortBy.vue renders the expected HTML structure, after search 1`] = `
<span
  id="sort-button-group"
>
  <span>
    Sort By: 
  </span>
   
  <button
    class="relevant-sort-btn active"
    value="relevance"
  >
    
        Relevance
    
  </button>
   
  <button
    class="newest-sort-btn"
    disabled="disabled"
    value="newest"
  >
    
        Newest
    
  </button>
</span>
`;

exports[`SortBy.vue renders the expected HTML structure, before search 1`] = `<!---->`;
